llvm/lib/Transforms/InstCombine/InstCombineCompares.cpp

```diff
@@ -7103,6 +7103,39 @@ Instruction *InstCombinerImpl::foldICmpCommutative(ICmpInst::Predicate Pred,
   if (Value *V = foldICmpWithLowBitMaskedVal(Pred, Op0, Op1, Q, *this))
     return replaceInstUsesWith(CxtI, V);
 
+  // Folding (X / Y) pred X => X ~pred 0 for some constant Y other than 0 or 1
```

Comment:
nit: `~pred` looks like the inverse predicate, not the swapped predicate, which are different things. I'd write this either as
```suggestion
  // Folding (X / Y) pred X => 0 pred X for some constant Y other than 0 or 1
```
or
```suggestion
  // Folding (X / Y) pred X => X swap(pred) 0 for some constant Y other than 0 or 1
```